test/no-for-loop-extra-variable.js

```diff
@@ -0,0 +1,325 @@
+				for(var i = 1, j = arr.length; j > i; i += 1) {
+					console.log('string');
+				}
+				console.log(j);
+			`,
+			errors: [error],
+			output: outdent`
+			var j = arr.length
+			for(var i = 1; j > i; i += 1) {
+				console.log('string');
+			}
+			console.log(j);
+		`
+		},
+		{
+			// J used in loop and after loop (let)
+			code: outdent`
+				for(let i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+				let j = 3;
+			`,
+			errors: [error],
+			output: outdent`
+			for(let i = 1, j = arr.length; j > i; i += 1) {
+				console.log(j);
+			}
+			let j = 3;
+		`
+		},
+		{
+			// J used in loop and after loop (var)
+			code: outdent`
+				for(var i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+				console.log(j);
+			`,
+			errors: [error],
+			output: outdent`
+			var j = arr.length
+			for(var i = 1; j > i; i += 1) {
+				console.log(j);
+			}
+			console.log(j);
+		`
+		},
+		{
+			// J used before loop only (let)
+			code: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			for(let i = 1, j = arr.length; j > i; i += 1) {
+					console.log('string');
+				}
+			}`,
+			errors: [error],
+			output: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			for(let i = 1; arr.length > i; i += 1) {
+					console.log('string');
+				}
+			}`
+		},
+		{
+			// J used before and in loop (let)
+			code: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			for(let i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+			}`,
+			errors: [error],
+			output: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			let j = arr.length
+			for(let i = 1; j > i; i += 1) {
+					console.log(j);
+				}
+			}`
+		},
+		{
+			// J used before and in loop (var)
+			code: outdent`
+			if(conditional){
+				var j = 5
+			}
+			else{	
+			for(var i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+			}`,
+			errors: [error],
+			output: outdent`
+			if(conditional){
+				var j = 5
+			}
+			else{	
+			var j = arr.length
+			for(var i = 1; j > i; i += 1) {
+					console.log(j);
+				}
+			}`
+		},
+		{
+			// J used before, in, and after loop (let)
+			code: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			for(let i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+			let j = 5
+			}`,
+			errors: [error],
+			output: outdent`
+			if(conditional){
+				let j = 5
+			}
+			else{	
+			for(let i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+			let j = 5
+			}`
+		},
+		{
+			// J used before, in, and after loop (var)
+			code: outdent`
+			if(conditional){
+				console.log(j)
+			}
+			else{	
+			for(var i = 1, j = arr.length; j > i; i += 1) {
+					console.log(j);
+				}
+			console.log(j)
+			}`,
+			errors: [error],
+			output: outdent`
+			if(conditional){
+				console.log(j)
+			}
+			else{	
+			var j = arr.length
+			for(var i = 1; j > i; i += 1) {
+					console.log(j);
+				}
```

Comment:
Many of these examples have messy code-style, which makes them hard to read. Would be good if you could fix that.